### PR TITLE
Update CSharpier to 0.30.3

### PR DESCRIPTION
**Pull request by pmachapman**
This PR:

 * Updates CSharpier to 0.30.3, which performs these changes:
    * Enforce trailing commas for multi-line object and collection initializer
    * Remove training commas for for singleline object and collection initializer
    * Fix blank line being added with lambda returning collection expression
    * Fix inconsistent Formatting for new() Operator Compared to Explicit Object Constructor
 * Updates .editorconfig to match CSharpier's suggested configuration
 * Enable collection expressions by default via `dotnet format`

If have broken these changes into separate commits in case you think collection expressions are a bad idea and I need to revert it (I personally prefer the syntax and use it when updating code or writing new code).

I figure now is the best time for a change like this, as a major update to the C# code base (SF-2900) has been merged.

- - -
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2912)

### CODE REVIEW

**Review by Nateowami — APPROVED** [review]
:lgtm: with two comments. I'm fine with merging; just wanted to leave some opinions.

> Reviewed 93 of 93 files at r1.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2912#-:-OEHJYwzBaMcfdMpEMV8:bx3qok2)* status: 92 of 93 files reviewed, 5 unresolved discussions (waiting on @pmachapman)

___
*[`src/SIL.XForge/DataAccess/MemoryRepository.cs` line 171 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2912#-OEH6AlfBYWrk6PJtyOb:-OEH6AlfBYWrk6PJtyOc:bdm4bxf) ([raw file](https://github.com/sillsdev/web-xforge/blob/86eaa350cd32b158936023ae6fcb09d061b23ab7/src/SIL.XForge/DataAccess/MemoryRepository.cs#L171)):*
> ```Smalltalk
>     public Task<int> DeleteAllAsync(Expression<Func<T, bool>> filter)
>     {
>         T[] entities = Query().Where(filter).ToArray();
> ```

Personally I think a method like `ToArray()` makes a lot more sense than using a spread operator because it's easier to follow a chain of methods, than syntax that alternates between chaining and whatever the opposite of chaining is (I consider `a().b().c()` to b easier to follow than `c(b(a()))`).
___
*[`src/SIL.XForge.Scripture/Services/ParatextService.cs` line 285 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2912#-OEHAIVzE8PLRZ6rGEP3:-OEHAIVzE8PLRZ6rGEP4:b-bk6mma) ([raw file](https://github.com/sillsdev/web-xforge/blob/8c28b6dc52972b75f7bcfbc09369738d9267d966/src/SIL.XForge.Scripture/Services/ParatextService.cs#L285)):*
> ```Smalltalk
>                 using ScrText scrText =
>                     ScrTextCollection.FindById(username, paratextId)
>                     ?? throw new Exception(
> ```

I'm not sure the `x = y() ?? throw new Z()` pattern is a good idea...

File: src/SIL.XForge/Realtime/RichText/Delta.cs [hunk 1]
```diff
@@ -512,18 +512,18 @@
     /// <summary>Provides methods to process ops for text that may have attribute changes</summary>
     private class DeltaOpsAttributeHelper
     {
-        private List<int> opLengths = new List<int>();
-        private List<JObject> originalDeltaOps = new List<JObject>();
-        private List<JObject> newDeltaOps = new List<JObject>();
+        private List<int> opLengths = [];
```
[the remark this answers is not in the document]
Comment (pmachapman) [reply src/SIL.XForge/Realtime/RichText/Delta.cs]: Done.

File: src/SIL.XForge/Realtime/RichText/Delta.cs [hunk 2]
```diff
@@ -512,18 +512,18 @@
...
+        private List<JObject> originalDeltaOps = [];
```
[the remark this answers is not in the document]
Comment (pmachapman) [reply src/SIL.XForge/Realtime/RichText/Delta.cs]: Done.

File: src/SIL.XForge/Realtime/RichText/Delta.cs [hunk 3]
```diff
@@ -512,18 +512,18 @@
...
+        private List<JObject> newDeltaOps = [];
```
[the remark this answers is not in the document]
Comment (pmachapman) [reply src/SIL.XForge/Realtime/RichText/Delta.cs]: Done.

**Review by pmachapman — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2912)* status: 92 of 93 files reviewed, 5 unresolved discussions (waiting on @github-advanced-security[bot] and @Nateowami)

___
*[`src/SIL.XForge/DataAccess/MemoryRepository.cs` line 171 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2912#-OEH6AlfBYWrk6PJtyOb:-OEHK2B41zs-WO81eDH_:b3l7in1) ([raw file](https://github.com/sillsdev/web-xforge/blob/86eaa350cd32b158936023ae6fcb09d061b23ab7/src/SIL.XForge/DataAccess/MemoryRepository.cs#L171)):*
<details><summary><i>Previously, Nateowami wrote…</i></summary><blockquote>

Personally I think a method like `ToArray()` makes a lot more sense than using a spread operator because it's easier to follow a chain of methods, than syntax that alternates between chaining and whatever the opposite of chaining is (I consider `a().b().c()` to b easier to follow than `c(b(a()))`).
</blockquote></details>

Yeah I was initially hostile to it, but since TypeScript uses the similar `[...otherArray]`, it has grown on me.
___
*[`src/SIL.XForge.Scripture/Services/ParatextService.cs` line 285 at r1](https://reviewable.io/reviews/sillsdev/web-xforge/2912#-OEHAIVzE8PLRZ6rGEP3:-OEHKHwj6ccFCyeO6j3Y:boc6nv3) ([raw file](https://github.com/sillsdev/web-xforge/blob/8c28b6dc52972b75f7bcfbc09369738d9267d966/src/SIL.XForge.Scripture/Services/ParatextService.cs#L285)):*
<details><summary><i>Previously, Nateowami wrote…</i></summary><blockquote>

I'm not sure the `x = y() ?? throw new Z()` pattern is a good idea...
</blockquote></details>

Yeah that was unexpected from `dotnet format`.

I believe it is more efficient in the generated MSIL code, as is it ensures that the `finally { scrText.Dispose(); }` which the `using` does in the background is not executed.

**Review by pmachapman — COMMENTED** [review]
> Dismissed @github-advanced-security[bot] from 3 discussions.
> *[Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/2912)* status: 91 of 93 files reviewed, 2 unresolved discussions (waiting on @Nateowami)

**Pull request merged**